### PR TITLE
Improve release version update

### PR DESCRIPTION
**Pull request by ferc**
## Details
- Add a select to pick which version to increment (major, minor or patch)
- Use `cut` to get the version parts
- Move the increment version at the beginning of the release job
- Consolidate the Git actions (increment version, tagging and pushing) into the same job to avoid possible race conditions

## Issues

Resolves #

### CODE REVIEW

**Review by liyaka — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -4,139 +4,139 @@ run-name: "Release from ${{github.ref_name}} by @${{ github.actor }}"
...
-      - set-version
     runs-on: ubuntu-latest
+    outputs:
+      new_version: ${{ steps.increment_version.outputs.new_version }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4.1.1
 
+      - name: Increment version
+        id: increment_version
```
Comment (liyaka) [inline .github/workflows/release.yaml:28]: i suggest to add
`shell: bash`

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -4,139 +4,139 @@ run-name: "Release from ${{github.ref_name}} by @${{ github.actor }}"
...
+          echo "$new_version" > version.txt
+
+      - name: Commit updated version file
+        run: |
+          git config --local user.email "github-actions@comet.com"
+          git config --local user.name "github-actions"
+          git add version.txt
+          git commit -m "Set version to ${{ steps.increment_version.outputs.new_version }}"
+
+      - name: Push new version.txt
```
Comment (liyaka) [inline .github/workflows/release.yaml:76]: you don't need this one

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -4,139 +4,139 @@ run-name: "Release from ${{github.ref_name}} by @${{ github.actor }}"
...
     with:
-      version: ${{needs.set-version.outputs.version}}
+      version: ${{ needs.create-git-tag.outputs.new_version }}
 
   release-sdk:
     needs:
-      - set-version
       - create-git-tag
-    uses: ./.github/workflows/build_and_publish_sdk.yaml
+    uses: ./.github/workflows/build_and_publish_sdk.yml
```
Comment (liyaka) [inline .github/workflows/release.yaml:102]: i see the name of the file is build_and_publish_sdk.yaml

**Review by liyaka — CHANGES_REQUESTED** [review]

**Comment by ferc** [comment]
I'll reopen it soon

**Pull request closed without merging**